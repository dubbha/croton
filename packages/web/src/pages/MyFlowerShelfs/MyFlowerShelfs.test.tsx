import React from 'react';
import { render } from '@testing-library/react';

jest.mock('components/Header', () => {
  const React = require('react');
  return {
    Header: () => <div>Header</div>
  };
});

jest.mock('components/Footer', () => {
  const React = require('react');
  return {
    Footer: () => <div>Footer</div>
  };
});

jest.mock('connected-react-router', () => ({
  push: (path: string) => ({ type: 'callHistoryMethod', payload: { path } })
}));

describe('pages/MyFlowerShelfs', () => {
  it('should render successfully if user is logged in', () => {
    jest.isolateModules(() => {
      const fn = jest.fn();
      jest.doMock('react-redux', () => ({
        useDispatch: () => fn,
        useSelector: () => ({
          token: 'TOKEN',
          firstName: 'FIRST_NAME',
          lastName: 'LAST_NAME',
          email: 'EMAIL'
        })
      }));

      const { MyFlowerShelfs } = require('./MyFlowerShelfs');

      const { container } = render(<MyFlowerShelfs />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should redirect to sing in if user is not logged in', () => {
    jest.isolateModules(() => {
      const fn = jest.fn();
      jest.doMock('react-redux', () => ({
        useDispatch: () => fn,
        useSelector: () => ({
          token: null,
          firstName: null,
          lastName: null,
          email: null
        })
      }));

      const { MyFlowerShelfs } = require('./MyFlowerShelfs');

      render(<MyFlowerShelfs />);

      expect(fn).toBeCalledWith({
        type: 'callHistoryMethod',
        payload: { path: '/signin' }
      });
    });
  });
});
