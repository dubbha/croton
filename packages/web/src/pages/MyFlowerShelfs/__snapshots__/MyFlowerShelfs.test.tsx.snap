// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/MyFlowerShelfs should render successfully if user is logged in 1`] = `
<div
  class="my-flower-shelfs"
>
  <h5>
    You don't have any flower pots shelfs yet.
  </h5>
</div>
`;
